server/plugin.go

```diff
@@ -225,6 +229,109 @@ func (p *Plugin) handleAdd(w http.ResponseWriter, r *http.Request) {
 	p.postReplyIfNeeded(addRequest.PostID, replyMessage, addRequest.Message)
 }
 
+type updateAPIRequest struct {
+	Message string `json:"message"`
+	SendTo  string `json:"send_to"`
+	PostID  string `json:"post_id"`
+}
+
+
+func (p *Plugin) handleUpdate(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	var updateRequest *updateAPIRequest
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&updateRequest)
+	if err != nil {
+		p.API.LogError("Unable to decode JSON err=" + err.Error())
+		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to decode JSON", err)
+		return
+	}
+
+	senderName := p.listManager.GetUserName(userID)
+
+	if updateRequest.SendTo == "" {
+		_, err = p.listManager.UpdateIssue(userID, updateRequest.Message, updateRequest.PostID)
+		if err != nil {
+			p.API.LogError("Unable to add issue err=" + err.Error())
+			p.handleErrorWithCode(w, http.StatusInternalServerError, "Unable to add issue", err)
+			return
+		}
+		p.trackUpdateIssue(userID, sourceWebapp, updateRequest.PostID != "")
+		replyMessage := fmt.Sprintf("@%s attached a todo to this thread", senderName)
+		p.postReplyIfNeeded(updateRequest.PostID, replyMessage, updateRequest.Message)
```

Comment:
Since I believe we should not be adding the todo to a thread on update, this will not be needed. My question would be: do we want to notify the thread if there are changes in the issue? @aaronrothschild ?

Reply:
0/5 Too spammy...

Reply:
I agree, probably too spammy.  I suggest an optional update to the thread IF the user provides a text update/reason.  But this is adding some complexity.  

Let's just say we won't update the post if the todo is updated for now.